python_functions_recursion.py

```diff
@@ -0,0 +1,27 @@
+from math import factorial
+from functools import lru_cache
+
+my_num = 8
+
+lru_cache(1000) #built-in cache. It should enhance performance. Idea is from Socratica https://www.youtube.com/watch?v=Qk0zUZW-U_M&t 
+def my_custom_factorial(a):
+    my_counter = a 
+    if a == 0:
+        return(1)
+    elif a > 9999:
+        print("Sorry, im not going to melt my processor. Go find a NASA computer.")
+    else: 
+        while my_counter >= 0: #To je prasárna, ten loop. 
```

Comment:
Jde to i jednoduseji. Ten While loop uplne nevim k cemu tam mate. Stacilo by (i s dobrym napadem na omezeni max poctu, napsat:

```
def my_custom_factorial(a):
    my_counter = a 
    if a == 0:
        return(1)
    elif a > 9999:
        print("Sorry, im not going to melt my processor. Go find a NASA computer.")
    else: 
        return a * (my_custom_factorial(a-1))
```